### DISCUSSION TITLE
About to try firebase as my backend. Any tips, suggestions, or anything under the sun?

### DISCUSSION THREAD

**Discussion by mikkodimapasoc** [vercel/next.js#17259] [Ideas]
Hiiii. First post in here. I`m about to try firebase as my backend for my next.js applications. Would probably use it too for authentication.  It would be my first time doing this(past experience was with REST Api using node and express), any suggestions, or tips or anything?

Stay safe!

**Comment by lfades** [comment]
Get started as soon as possible 👍 

Firebase is a very good service and Next.js can take care of anything you may want to build with it.

**Reply by leerob** [reply]
+1

- Check out these React hooks the Firebase team is working on: https://github.com/FirebaseExtended/reactfire
- If you need to use Firebase inside `getStaticProps`/`getServerSideProps`, these run in the Node context. You'll need to connect to Firebase Admin, which is different than client-side Firebase. 
- If you do end up adding Firebase Admin and you're stuck adding environment variables with Vercel, [see this comment](https://github.com/leerob/fastfeedback/pull/6#issuecomment-641993945)

Here's two example repos that might help for reference:

- https://github.com/leerob/nextjs-vercel-firebase
- https://github.com/leerob/fastfeedback

**Reply by mikkodimapasoc** [reply]
Thank you guys!
Stay safe!
